### PR TITLE
Update `.travis.yml` to run on OSX.

### PR DESCRIPTION
**Pull request by NTaylorMullen**
- Added `MONO_THREADS_PER_CPU=2000` environment variable to ensure `dnu restore` completes successfully.

**Note:** It looks like OSX is currently failing so that will need to be investigated.

### CODE REVIEW

**Comment by BrennanConroy** [comment]
So all the failures are because the file path we are looking for doesn't have /private/ in the front. Not sure why yet.

**Comment by davidfowl** [comment]
We should fix the failures in @NTaylorMullen 's branch before merging this in.

**Comment by BrennanConroy** [comment]
Idea:
In Microsoft.Dnx.CommonTestUtils the CreateTempDir function we save the current directory, then set the current directory to the created temp directory, then set the temp directory string to the value from get current directory, and reset the current directory. Tested it locally on a Mac for the Microsoft.Dnx.Tooling.FunctionTests and it worked

**Comment by NTaylorMullen** [comment]
:+1:

File: test/Microsoft.Dnx.CommonTestUtils/TestUtils.cs [hunk 1]
```diff
@@ -30,6 +30,13 @@ public static DisposableDir CreateTempDir()
         {
             var tempDirPath = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName());
             Directory.CreateDirectory(tempDirPath);
+
+            // Resolves issues on Mac where GetTempPath gives /var and GetCurrentDirectory gives /private/var
+            var currentDirectory = Directory.GetCurrentDirectory();
```
Comment (davidfowl) [inline test/Microsoft.Dnx.CommonTestUtils/TestUtils.cs]: Make sure this only runs on mac

**Comment by davidfowl** [comment]
:shipit:

**Comment by BrennanConroy** [comment]
fb88afaeed702b1af730f49b49f696b3f8e8b68f

**Pull request closed without merging**